### ISSUE TITLE
--binarylog option broken

### ISSUE THREAD

**Issue by sbomer** [dotnet/format#2094]
`--binarylog` no longer seems to work. I bisected dotnet-format, and it was broken by https://github.com/dotnet/format/commit/f58bc29a159f43a71e6ac21fd2734f5d10807a72.

This related issue is mentioned: https://github.com/dotnet/roslyn/issues/70674.
According to https://github.com/dotnet/roslyn/discussions/71950#discussion-6174863, broken by https://github.com/dotnet/roslyn/pull/70469 as part of an intentional change to the API.

**Comment by JoeRobich** [comment]
Let me start by saying that a fix is planned for this break in behavior. Although, depending on how it is resolved in Roslyn, there may be a change in how it is used in dotnet-format. 

However, since binary logs are a pretty big hammer, I think it is good to touch base and ensure we are providing the right feature here. These questions are open to anyone using the `--binarylog` option.

How are you using the generated log?
What information is it providing for you?
What questions are you answering with it?

**Comment by sharwell** [comment]
> However, since binary logs are a pretty big hammer, I think it is good to touch base and ensure we are providing the right feature here.

I'm not sure I agree with this. In my experience, binary logs are _the only_ hammer that matters when looking at build issues.

**Comment by jasonmalinowski** [comment]
@JoeRobich We got this working again, right?

**Comment by JoeRobich** [comment]
Yes, This was fixed in https://github.com/dotnet/roslyn/pull/80120
